lib/data/json.js

```diff
@@ -244,7 +244,7 @@ const submissionToOData = (fields, table, { xml, encHasData, submission, submitt
               dataPtr[name].properties = { accuracy };
           }
         } else if ((type === 'geotrace') || (type === 'geoshape')) {
-          const pointStrs = text.split(';');
+          const pointStrs = text.replace(/;$/, '').split(';');
```

Comment:
is `; ` etc not possible ?

Reply:
actually i'm a little confused how the failure case occurs in the first place. if there is a trailing `;` the lat/lon parsing below should fail and it should not output the point, no?

Reply:
You mean could there be trailing spaces? No client that I know of currently produces them but I'm happy to add a `trim`.

Reply:
I didn't add a test for the `trim` since it feels like an "extra". If you prefer that I do, do you think it's ok to slip it into one of the existing cases or should it be a dedicated test?

Didn't see the other comment before for some reason. 

> if there is a trailing ; the lat/lon parsing below should fail and it should not output the point, no?

JS is happy to take null and undefined values and make them into floats like `undefined` and `NaN`. I got the same results as in the issue (#282) when running the tests without the `replace`. Did you expect a crash? There's no explicit existence check for lat or lon. In the case of malformed points like only a single float, you'll end up with something like `37 NaN`.

Reply:
ah. the NaN bailout logic didn't get copied down.

Reply:
It feels strange to me to produce a well-formed trace/shape from a bad input instead of including a hint that something went wrong. The ways to get to bad input are a rogue client, a bad default, or a faulty form expression that generates a trace/shape. I suppose they're not super likely. Anyway, fine by me to go with your approach.

Reply:
hmm? i'm not sure how my approach is any different in this regard than yours?

Reply:
i just don't like text manip. it's really expensive.

Reply:
Fair about expense. The difference is that previously malformed points within a trace/shape would be identifiable because they'd include a `NaN`. Now they'll just be skipped.

Reply:
oh. huh. i would have instinctively considered that a bug. since it’s not to spec. but i guess that makes sense.